pkg/internal/server/grpc/validators.go

```diff
@@ -17,42 +17,102 @@ limitations under the License.
 package grpc
 
 import (
+	"context"
+	"fmt"
+
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/kubernetes-csi/external-snapshot-metadata/pkg/api"
+	snapshot "github.com/kubernetes-csi/external-snapshotter/client/v6/clientset/versioned"
 )
 
-func ValidateGetMetadataAllocatedRequest(req *api.GetMetadataAllocatedRequest) error {
+func (s *Server) ValidateGetMetadataAllocatedRequest(ctx context.Context, req *api.GetMetadataAllocatedRequest) (string, error) {
 	if len(req.GetSecurityToken()) == 0 {
-		return status.Errorf(codes.InvalidArgument, msgInvalidArgumentSecurityTokenMissing)
+		return "", status.Errorf(codes.InvalidArgument, msgInvalidArgumentSecurityTokenMissing)
 	}
 
 	if len(req.GetNamespace()) == 0 {
-		return status.Errorf(codes.InvalidArgument, msgInvalidArgumentNamespaceMissing)
+		return "", status.Errorf(codes.InvalidArgument, msgInvalidArgumentNamespaceMissing)
 	}
 
 	if len(req.GetSnapshotName()) == 0 {
-		return status.Errorf(codes.InvalidArgument, msgInvalidArgumentSnaphotNameMissing)
+		return "", status.Errorf(codes.InvalidArgument, msgInvalidArgumentSnaphotNameMissing)
+	}
+
+	snapshotHandle, driver, err := getVolSnapshotInfo(ctx, s.snapshotClient(), req.Namespace, req.SnapshotName)
+	if err != nil {
+		return "", status.Errorf(codes.Unavailable, msgUnavailableVolumeSnapshotNotReady)
```

Comment:
`getVolumeSnapshotInfo` should return a properly formatted `status` error.

Reply:
I kept `getVolumeSnapshotInfo` as a generic helper method. But since we don't have any other consumer, I think we can do it